x-pack/plugins/security_solution/public/common/containers/source/index.tsx

```diff
@@ -228,37 +238,88 @@ export const useIndexFields = (sourcererScopeName: SourcererScopeName) => {
     [dispatch, sourcererScopeName]
   );
 
+  const getSignalIndexNameSelector = useMemo(
+    () => sourcererSelectors.signalIndexNameSelector(),
+    []
+  );
+  const signalIndexNameSelector = useDeepEqualSelector(getSignalIndexNameSelector);
+
   const indexFieldsSearch = useCallback(
-    (indicesName) => {
+    (selectedDataViewId: string, newSignalsIndex?: string) => {
       const asyncSearch = async () => {
         abortCtrl.current = new AbortController();
         setLoading(true);
         searchSubscription$.current = data.search
-          .search<IndexFieldsStrategyRequest, IndexFieldsStrategyResponse>(
-            { indices: indicesName, onlyCheckIfIndicesExist: false },
+          .search<IndexFieldsStrategyRequest<'dataView'>, IndexFieldsStrategyResponse>(
+            {
+              dataViewId: selectedDataViewId,
+              onlyCheckIfIndicesExist: false,
+            },
             {
               abortSignal: abortCtrl.current.signal,
               strategy: 'indexFields',
             }
           )
           .subscribe({
             next: (response) => {
+              // TODO: Steph/sourcerer needs better tests
               if (isCompleteResponse(response)) {
-                const stringifyIndices = response.indicesExist.sort().join();
-                dispatch(
-                  sourcererActions.setSource({
-                    id: sourcererScopeName,
-                    payload: {
-                      browserFields: getBrowserFields(stringifyIndices, response.indexFields),
-                      docValueFields: getDocValueFields(stringifyIndices, response.indexFields),
-                      errorMessage: null,
-                      id: sourcererScopeName,
-                      indexPattern: getIndexFields(stringifyIndices, response.indexFields),
-                      indicesExist: response.indicesExist.length > 0,
-                      loading: false,
-                    },
-                  })
+                const signalIndexName = signalIndexNameSelector
+                  ? `${signalIndexNameSelector}-*`
```

Comment:
why do we need to do that? we never had to do that before

Reply:
we need it for the index pattern or we dont get fields, get fields for wildcard needs a wildcard

Reply:
I'm trying to get a screenshot now but can't reproduce field differences. but this is why our cypress test was failing, was pairing with @MadameSheema  and she saw it too. https://github.com/elastic/kibana/pull/111461/files/bed0e0b709de0cd48784eab5b79058227c067082#r720268892